specs/conductor.md

```diff
@@ -0,0 +1,58 @@
+
+In the second case, batches of blocks are received from the DA later, filtered for that rollup, and those blocks are used to set their corresponding blocks on the rollup as finalized.
+
+The exact terminology that a rollup uses for its fork choice rules is up to its implementation. For example, Geth uses `head`, `safe`, and `final`. The conductor uses `head`, `soft`, and `firm`. The fork choice options are mapped with Geth in the following way:
+ - `head` -> `head`
+ - `soft` -> `safe`
+ - `firm` -> `final`
+
+The conductor has no knowledge that Geth uses the above options, it simply sends a `CommitmentState` message to the rollup, and the rollup does the rest.
+
+## Architecture
+The architecture of the conductor is inspired by the [Actor Model](https://en.wikipedia.org/wiki/Actor_model) with the actors within the conductor being the `Driver`, `Reader`, and `Executor`. Each actor operates concurrently and communicates with the other actors by passing messages. The conductor is written in Rust and utilized the Tokio runtime to achieve this.
+
+![Conductor Architecture](assets/conductor-architecture.png)
+
+The responsibilities of each module are as follows:
+### Driver
+ - Top level coordinator that runs and manages all the sub-components necessary for the Conductor
+ - Creates the `Reader` and `Executor` actors on startup
+ - Creates a gossip network for receiving data from the sequencer network
+     - The gossip network uses the [astria-gossipnet](https://github.com/astriaorg/astria/tree/main/crates/astria-gossipnet)
+ - Runs an event loop that handles receiving `DriverCommand`s and messages from the gossip network
+ - Passes sequencer blocks received from the gossip network to the `Executor`
+
+### Reader
+ - Creates a `CelestiaClient` using the Celestia client implementation from `astria-sequencer-relayer` to communicating with the DA layer
+ - Creates a `TendermintClient` which is used when validating blocks from the DA layer
+ - Runs an event loop that handles receiving `ReaderCommand`s that drives data retrieval from the DA layer
+ - Passes the blocks it receives to the `Executor`
+
+### Executor
+ - Runs an event loop that handles receiving `ExecutorCommand`s from both the Driver and Reader
+ - Filters out transactions by their rollup namespace and sends them to the rollup for execution (`soft` commits)
+ - Catalogs and matches the block hashes received from the gossip network and the DA by rollup namespace to send `firm` commits to the rollup
+ - Blocks are sent to the execution layer using [Astria’s GRPC Execution client interface](https://buf.build/astria/astria/docs/main:astria.execution.v1)
+     - Rollups utilizing Astria must implement this interface
```

Comment:
This isn't strictly true, only that it's required to utilize conductor, and conductor makes it easier to integrate.

Reply:
updated to say Conductor instead of just Astria